### PR TITLE
v7.1.0

### PR DESCRIPTION
**Pull request by michelengelen**
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

---

- [x] Compare the last tag with the branch upon which you want to release.
- [x] Clean the generated changelog.
- [x] Update the root `package.json`'s version.
- [x] Update the versions of the other `package.json` files and of the dependencies.
- [x] Open PR with changes and wait for review and green CI.
- [x] Merge PR once CI is green, and it has been approved.

### Release the packages

- [ ] Checkout the last version of the working branch
- [ ] `yarn && yarn release:build`
- [ ] `yarn release:publish`
- [ ] `yarn release:tag`

### Publishing

- [ ] Publish v7 docs
- [ ] Publish GitHub release

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,72 @@
...
+- 🚀 Fix tooltip causing PieChart to crash on data change (#12571) @Rishi556
+- 🌍 Improve Chinese (Hong Kong) (zh-HK) and Italian (it-IT) locale for the Data Grid
+- 🐞 Bugfixes
+- 📚 Documentation improvements
+
+### Data Grid
+
+#### `@mui/x-data-grid@7.0.1`
+
+- [DataGrid] Add `resizeThrottleMs` prop (#12556) @romgrk
```
Comment (flaviendelangle) [inline CHANGELOG.md:20]: Technically speaking this is a new feature and could justify a minor :laughing:

Comment (michelengelen) [reply CHANGELOG.md:20]: OK, can change that ... makes sense

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,73 @@
...
+
+#### `@mui/x-data-grid-premium@7.1.0` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')
+
+Same changes as in `@mui/x-data-grid-pro@7.1.0`.
+
+### Date and Time Pickers
+
+#### `@mui/x-date-pickers@7.1.0`
+
+- [pickers] Prepare compatibility with `@mui/zero-runtime` (stop using `ownerState` in `styled`) (#12003) @flaviendelangle
```
Comment (flaviendelangle) [inline CHANGELOG.md:40]:
```suggestion
- [fields] Fix placeholder override] (#12589) @flaviendelangle
- [pickers] Prepare compatibility with `@mui/zero-runtime` (stop using `ownerState` in `styled`) (#12003) @flaviendelangle
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,74 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 7.1.0
+
+_Mar 28, 2024_
+
+We'd like to offer a big thanks to the 10 contributors who made this release possible. Here are some highlights ✨:
+
+- 🚀 Fix tooltip causing PieChart to crash on data change (#12571) @Rishi556
```
Comment (flaviendelangle) [inline CHANGELOG.md:12]: Not sure this is worth a highlight, I don"t remember us highlighting any bug fix

**Review by alexfauquette — APPROVED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (flaviendelangle) [reply CHANGELOG.md]: IMHO, we could highlight when it's a super major bug that a good portion of the community is waiting for.
But this should be an exception, we don't want to put to much light on the existence of bugs :laughing:

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,75 @@
...
+
+Same changes as in `@mui/x-data-grid-pro@7.1.0`.
+
+### Date and Time Pickers
+
+#### `@mui/x-date-pickers@7.1.0`
+
+- [fields] Fix placeholder override (#12589) @flaviendelangle
+- [pickers] Prepare compatibility with `@mui/zero-runtime` (stop using `ownerState` in `styled`) (#12003) @flaviendelangle
+- [DateRangePicker] Fix selection behavior with single input field when `readonly` (#12593) @LukasTy
```
[the remark this answers is not in the document]
Comment (LukasTy) [reply CHANGELOG.md:42]: We do, but this should be listed for `@mui/x-date-pickers-pro`. 😉

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,75 @@
...
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 7.1.0
+
+_Mar 28, 2024_
+
+We'd like to offer a big thanks to the 10 contributors who made this release possible. Here are some highlights ✨:
+
+- 🚀 Add `resizeThrottleMs` prop (#12556) @romgrk
+- 🌍 Improve Chinese (Hong Kong) (zh-HK) and Italian (it-IT) locale for the Data Grid
```
Comment (LukasTy) [inline CHANGELOG.md:13]:
```suggestion
- 🌍 Improve Chinese (Hong Kong) (zh-HK) and Italian (it-IT) locale on the Pickers
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,6 +3,75 @@
...
+### Data Grid
+
+#### `@mui/x-data-grid@7.1.0`
+
+- [DataGrid] Add `resizeThrottleMs` prop (#12556) @romgrk
+- [DataGrid] Do not publish `rowEditStop` event if row has fields with errors (#11383) @cherniavskii
+- [DataGrid] Fix bug in suspense (#12553) @romgrk
+- [DataGrid] Fix missing class name in the `GridToolbarQuickFilter` component (#12484) @jhawkins11
+- [l10n] Improve Chinese (Hong Kong) (zh-HK) locale (#12547) @samchiu90
+- [l10n] Improve Italian (it-IT) locale (#12549) @antomanc
```
Comment (LukasTy) [inline CHANGELOG.md:26]:
```suggestion
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -3,6 +3,75 @@
...
+
+#### `@mui/x-data-grid-premium@7.1.0` [![premium](https://mui.com/r/x-premium-svg)](https://mui.com/r/x-premium-svg-link 'Premium plan')
+
+Same changes as in `@mui/x-data-grid-pro@7.1.0`.
+
+### Date and Time Pickers
+
+#### `@mui/x-date-pickers@7.1.0`
+
+- [fields] Fix placeholder override (#12589) @flaviendelangle
```
Comment (LukasTy) [inline CHANGELOG.md:40]:
```suggestion
- [fields] Fix placeholder override (#12589) @flaviendelangle
- [l10n] Improve Chinese (Hong Kong) (zh-HK) locale (#12547) @samchiu90
- [l10n] Improve Italian (it-IT) locale (#12549) @antomanc
```

File: CHANGELOG.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (LukasTy) [reply CHANGELOG.md:42]:
```suggestion
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -3,6 +3,75 @@
...
+- [pickers] Prepare compatibility with `@mui/zero-runtime` (stop using `ownerState` in `styled`) (#12003) @flaviendelangle
+- [DateRangePicker] Fix selection behavior with single input field when `readonly` (#12593) @LukasTy
+
+#### `@mui/x-date-pickers-pro@7.1.0` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')
+
+Same changes as in `@mui/x-date-pickers@7.1.0`.
```
Comment (LukasTy) [inline CHANGELOG.md:46]:
```suggestion
Same changes as in `@mui/x-date-pickers@7.1.0`, plus:

- [DateRangePicker] Fix selection behavior with single input field when `readOnly` (#12593) @LukasTy
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -3,6 +3,75 @@
...
+
+#### `@mui/x-tree-view@7.1.0`
+
+- [TreeView] Do not use outdated version of the state to compute new label first char in `RichTreeView` (#12512) @flaviendelangle
+
+### Docs
+
+- [docs] Add example to add a second icon next to the field's opening button (#12524) @flaviendelangle
+- [docs] Add missing note to Data Grid migration guide (#12557) @romgrk
+- [docs] Chart title for SEO (#12545) @oliviertassinari
```
Comment (LukasTy) [inline CHANGELOG.md:64]:
```suggestion
- [docs] Fix Charts title for SEO (#12545) @oliviertassinari
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -3,6 +3,76 @@
...
+
+- [TreeView] Do not use outdated version of the state to compute new label first char in `RichTreeView` (#12512) @flaviendelangle
+
+### Docs
+
+- [docs] Add example to add a second icon next to the field's opening button (#12524) @flaviendelangle
+- [docs] Add missing note to Data Grid migration guide (#12557) @romgrk
+- [docs] Fix Charts title for SEO (#12545) @oliviertassinari
+- [docs] Fix small typo (#12558) @diogoparente
+- [docs] Reduce noise in migration docs side navigation (#12552) @cherniavskii
```
Comment (MBilalShafi) [inline CHANGELOG.md:67]: I just merged a PR.
```suggestion
- [docs] Improve codemod related documentation (#12582) @MBilalShafi
- [docs] Reduce noise in migration docs side navigation (#12552) @cherniavskii
```

**Review by MBilalShafi — APPROVED** [review]

**Pull request merged**